sdk/core/Azure.Core/src/Shared/RequestContentHelper.cs

```diff
@@ -62,6 +64,12 @@ public static RequestContent FromEnumerable(IEnumerable<BinaryData> enumerable)
             return content;
         }
 
+#if !NET5_0_OR_GREATER
+        // RequiresUnreferencedCode in net5.0 doesn't have AttributeTargets.Class as a target, but it was added in net6.0
+        // The call to JsonSerializer.Serialize is not compatible with trimming, but is only used in NET 5 and below, so don't warn otherwise.
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode(SerializationTrimmingMessage)]
+        [System.Diagnostics.CodeAnalysis.RequiresDynamicCode(SerializationTrimmingMessage)]
+#endif
```

Comment:
Why do we need these - I would think it's the same problem as the other method in this file (the OTel assembly not targetting net6 yet)

Reply:
removed - missed those 2